packages/form-js-viewer/src/render/components/form-fields/FilePicker.js

```diff
@@ -71,24 +80,29 @@ export function FilePicker(props) {
             onChange({
               value: null,
             });
+            fileRegistry.deleteFiles(filesKey);
+            setSelectedFiles([]);
+
             return;
           }
 
           const files = Array.from(input.files);
 
           onChange({
-            value: `${id}_value_key`,
+            value: filesKey,
           });
 
+          fileRegistry.setFiles(filesKey, files);
+
           setSelectedFiles(files);
         }}
       />
       <div className="fjs-filepicker-container">
         <button
           type="button"
-          disabled={disabled}
+          disabled={disabled || readonly || fileRegistry === null}
```

Comment:
Just FYI, at the moment it's still possible to submit a file in the editor. I guess you're planning on adding a custom editor renderer for this anyways?

Reply:
If not, needs fixing for sure. I don't know if it's related to the label or something but when I click around on it in the editor, it opens the window, which can lead to an exception.

Reply:
As I mentioned through Slack I couldn't reproduce it, maybe we can check it together tomorrow